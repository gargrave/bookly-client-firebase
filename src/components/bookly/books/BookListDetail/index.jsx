// @flow
import React from 'react';
import { bool, func, shape, string } from 'prop-types';

import type { Book } from '../../../../constants/flowtypes';

import Card from '../../../common/Card';
import CardTextLine from '../../../common/Card/CardTextLine';

type Props = {
  book: Book,
  onClick: Function,
  showAuthor?: boolean,
};

const BookListDetail = ({
  book,
  onClick,
  showAuthor,
}: Props) => {
  const {
    author,
    title,
  } = book;
  const authorName = `${author.firstName} ${author.lastName}`;

  return (
    <Card
      classes={[
        'list-detail-card',
        'book-list-detail-card',
      ]}
      onClick={onClick}
    >
      <CardTextLine bold text={title} />
      {showAuthor && <CardTextLine text={authorName} />}
    </Card>
  );
};

BookListDetail.propTypes = {
  book: shape({
    author: shape({
      firstName: string.isRequired,
      lastName: string.isRequired,
    }),
    title: string.isRequired,
  }).isRequired,
  onClick: func.isRequired,
  showAuthor: bool,
};

export default BookListDetail;
