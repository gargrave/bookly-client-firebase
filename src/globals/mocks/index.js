import { authorMocks } from './data/authorMocks';
import { bookBucketMocks } from './data/bookBucketMocks';
import { bookMocks } from './data/bookMocks';
import { profileMocks } from './data/profileMocks';
import { userMocks } from './data/userMocks';

export {
  authorMocks,
  bookBucketMocks,
  bookMocks,
  profileMocks,
  userMocks,
};
