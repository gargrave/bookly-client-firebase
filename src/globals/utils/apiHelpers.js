// @flow
import type { FbDocRef } from '../../modules/core/flowtypes';

async function getDocRef(
  db: any,
  table: string,
  id: string,
): FbDocRef {
  return db.collection(table).doc(id);
}

export {
  getDocRef,
};
