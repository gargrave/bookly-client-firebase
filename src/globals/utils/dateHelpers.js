export const toUnixTimestamp = (date, defaultValue) => {
  if (!date) {
    if (defaultValue === null || defaultValue === undefined) {
      return Date.now();
    }
    return defaultValue;
  }

  if (date._isAMomentObject) {
    return date.unix();
  }

  if (typeof date === 'number') {
    return date;
  }

  return defaultValue;
};

export const toUnixMs = (date) =>
  (date && date * 1000) || 0;
