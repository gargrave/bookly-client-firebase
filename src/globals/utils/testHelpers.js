import React from 'react';
import { shallow } from 'enzyme';

export class ComponentBuilder {
  constructor(component, defaultProps) {
    this.component = component;
    this.defaultProps = defaultProps;
  }

  shallowGetComponent(extraProps = {}) {
    const props = {
      ...this.defaultProps,
      ...extraProps,
    };
    return shallow(<this.component {...props} />);
  }
}
