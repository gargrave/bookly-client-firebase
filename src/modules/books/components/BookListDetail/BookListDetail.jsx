// @flow
import React from 'react';
import { bool, func, shape, string } from 'prop-types';

import type { Book } from '../../flowtypes';

import Card from '../../../common/components/Card/Card';

type Props = {
  book: Book,
  onClick: Function,
  showAuthor?: boolean,
};

const BookListDetail = ({
  book,
  onClick,
  showAuthor,
}: Props) => {
  const { author, title } = book;
  const authorName = `${author.firstName} ${author.lastName}`;

  return (
    <Card
      classes={[
        'list-detail-card',
        'book-list-detail-card',
      ]}
      onClick={onClick}
    >
      <Card.TextLine bold text={title} />
      {showAuthor && <Card.TextLine text={authorName} />}
    </Card>
  );
};

BookListDetail.propTypes = {
  book: shape({
    author: shape({
      firstName: string.isRequired,
      lastName: string.isRequired,
    }),
    title: string.isRequired,
  }).isRequired,
  onClick: func.isRequired,
  showAuthor: bool,
};

export default BookListDetail;
