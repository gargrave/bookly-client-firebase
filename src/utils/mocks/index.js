import authorMocks from './data/authors';
import userMocks from './data/users';

export {
  authorMocks,
  userMocks,
};
