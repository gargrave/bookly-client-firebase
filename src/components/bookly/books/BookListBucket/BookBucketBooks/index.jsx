// @flow
import React from 'react';
import { array, bool, func, shape, string } from 'prop-types';

import type { Book,BookBucket } from '../../../../../constants/flowtypes/';

import { buildClasses } from '../../../../../utils/cssHelpers';

import BookListDetail from '../../BookListDetail/';

type Props = {
  bucket: BookBucket,
  onBookClick: Function,
  showAuthors?: boolean,
};

function booksFromBucket(
  bucket: BookBucket,
  onBookClick: Function,
  showAuthors?: boolean,
) {
  return bucket.books.map((book: Book) => {
    return (
      <BookListDetail
        book={book}
        key={book.id}
        onClick={onBookClick.bind(null, book.id)}
        showAuthor={showAuthors}
      />
    );
  });
};

function BookBucketBookList({
  bucket,
  onBookClick,
  showAuthors,
}: Props) {
  return (
    <div className={buildClasses(['book-bucket__book-list'])}>
      {booksFromBucket(bucket, onBookClick, showAuthors)}
    </div>
  );
}

BookBucketBookList.propTypes = {
  bucket: shape({
    author: string,
    books: array,
  }).isRequired,
  onBookClick: func.isRequired,
  showAuthors: bool,
};

export default BookBucketBookList;
