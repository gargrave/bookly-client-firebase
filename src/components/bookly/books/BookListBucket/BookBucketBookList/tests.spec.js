import React from 'react';
import { shallow } from 'enzyme';

import { bookBucketMocks } from '../../../../../utils/mocks/';

import BookBucketBooks from './';
import BookListDetail from '../../BookListDetail/';

const defaultProps = Object.freeze({
  bucket: bookBucketMocks[0],
  onBookClick: jest.fn(),
});

function getComponent(extraProps = {}) {
  const props = Object.assign({}, defaultProps, extraProps);
  return shallow(<BookBucketBooks {...props} />);
}

describe('BookBucketBooks', () => {
  let component;

  it('matches the snapshot', () => {
    component = getComponent();
    expect(component).toMatchSnapshot();
  });

  it('renders correctly', () => {
    component = getComponent();
    const parentClass = '.bookly-book-bucket__book-list';
    expect(component.find(parentClass).length).toEqual(1);
  });

  it ('renders the correct number of BookListDetail children', () => {
    const bookCount = bookBucketMocks[0].books.length;
    expect(component.find(BookListDetail).length).toEqual(bookCount);
  });
});
