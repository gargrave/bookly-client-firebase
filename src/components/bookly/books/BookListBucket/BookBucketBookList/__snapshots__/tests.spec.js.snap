// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookBucketBooks matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BookBucketBookList
    bucket={
      Object {
        "author": "Aldous Huxley",
        "books": Array [
          Object {
            "author": Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            },
            "created": 2018-01-14T01:36:26.220Z,
            "id": "sv3hmbefpc95MRXQsTHr",
            "title": "Brave New World",
            "updated": 2018-01-14T01:36:26.220Z,
          },
          Object {
            "author": Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            },
            "created": 2018-01-15T01:36:26.220Z,
            "id": "Hmh9eR3rs5bMvsTXfpcQ",
            "title": "Island",
            "updated": 2018-01-15T01:36:26.220Z,
          },
          Object {
            "author": Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            },
            "created": 2018-01-16T01:36:26.220Z,
            "id": "eRpcQ5H3rsTXfmh9bMvs",
            "title": "The Doors of Perception",
            "updated": 2018-01-16T01:36:26.220Z,
          },
        ],
      }
    }
    onBookClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <BookListDetail
          book={
            Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              },
              "created": 2018-01-14T01:36:26.220Z,
              "id": "sv3hmbefpc95MRXQsTHr",
              "title": "Brave New World",
              "updated": 2018-01-14T01:36:26.220Z,
            }
          }
          onClick={[Function]}
          showAuthor={false}
        />,
        <BookListDetail
          book={
            Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              },
              "created": 2018-01-15T01:36:26.220Z,
              "id": "Hmh9eR3rs5bMvsTXfpcQ",
              "title": "Island",
              "updated": 2018-01-15T01:36:26.220Z,
            }
          }
          onClick={[Function]}
          showAuthor={false}
        />,
        <BookListDetail
          book={
            Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              },
              "created": 2018-01-16T01:36:26.220Z,
              "id": "eRpcQ5H3rsTXfmh9bMvs",
              "title": "The Doors of Perception",
              "updated": 2018-01-16T01:36:26.220Z,
            }
          }
          onClick={[Function]}
          showAuthor={false}
        />,
      ],
      "className": "bookly-book-bucket__book-list",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "sv3hmbefpc95MRXQsTHr",
        "nodeType": "function",
        "props": Object {
          "book": Object {
            "author": Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            },
            "created": 2018-01-14T01:36:26.220Z,
            "id": "sv3hmbefpc95MRXQsTHr",
            "title": "Brave New World",
            "updated": 2018-01-14T01:36:26.220Z,
          },
          "onClick": [Function],
          "showAuthor": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "Hmh9eR3rs5bMvsTXfpcQ",
        "nodeType": "function",
        "props": Object {
          "book": Object {
            "author": Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            },
            "created": 2018-01-15T01:36:26.220Z,
            "id": "Hmh9eR3rs5bMvsTXfpcQ",
            "title": "Island",
            "updated": 2018-01-15T01:36:26.220Z,
          },
          "onClick": [Function],
          "showAuthor": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "eRpcQ5H3rsTXfmh9bMvs",
        "nodeType": "function",
        "props": Object {
          "book": Object {
            "author": Object {
              "bookCount": 1,
              "created": 2017-07-01T01:16:43.881Z,
              "firstName": "Aldous",
              "id": "oiw223h3qndlksadffl3ksdf",
              "lastName": "Huxley",
              "updated": 2017-07-01T01:16:43.881Z,
            },
            "created": 2018-01-16T01:36:26.220Z,
            "id": "eRpcQ5H3rsTXfmh9bMvs",
            "title": "The Doors of Perception",
            "updated": 2018-01-16T01:36:26.220Z,
          },
          "onClick": [Function],
          "showAuthor": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <BookListDetail
            book={
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-14T01:36:26.220Z,
                "id": "sv3hmbefpc95MRXQsTHr",
                "title": "Brave New World",
                "updated": 2018-01-14T01:36:26.220Z,
              }
            }
            onClick={[Function]}
            showAuthor={false}
          />,
          <BookListDetail
            book={
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-15T01:36:26.220Z,
                "id": "Hmh9eR3rs5bMvsTXfpcQ",
                "title": "Island",
                "updated": 2018-01-15T01:36:26.220Z,
              }
            }
            onClick={[Function]}
            showAuthor={false}
          />,
          <BookListDetail
            book={
              Object {
                "author": Object {
                  "bookCount": 1,
                  "created": 2017-07-01T01:16:43.881Z,
                  "firstName": "Aldous",
                  "id": "oiw223h3qndlksadffl3ksdf",
                  "lastName": "Huxley",
                  "updated": 2017-07-01T01:16:43.881Z,
                },
                "created": 2018-01-16T01:36:26.220Z,
                "id": "eRpcQ5H3rsTXfmh9bMvs",
                "title": "The Doors of Perception",
                "updated": 2018-01-16T01:36:26.220Z,
              }
            }
            onClick={[Function]}
            showAuthor={false}
          />,
        ],
        "className": "bookly-book-bucket__book-list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "sv3hmbefpc95MRXQsTHr",
          "nodeType": "function",
          "props": Object {
            "book": Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              },
              "created": 2018-01-14T01:36:26.220Z,
              "id": "sv3hmbefpc95MRXQsTHr",
              "title": "Brave New World",
              "updated": 2018-01-14T01:36:26.220Z,
            },
            "onClick": [Function],
            "showAuthor": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "Hmh9eR3rs5bMvsTXfpcQ",
          "nodeType": "function",
          "props": Object {
            "book": Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              },
              "created": 2018-01-15T01:36:26.220Z,
              "id": "Hmh9eR3rs5bMvsTXfpcQ",
              "title": "Island",
              "updated": 2018-01-15T01:36:26.220Z,
            },
            "onClick": [Function],
            "showAuthor": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "eRpcQ5H3rsTXfmh9bMvs",
          "nodeType": "function",
          "props": Object {
            "book": Object {
              "author": Object {
                "bookCount": 1,
                "created": 2017-07-01T01:16:43.881Z,
                "firstName": "Aldous",
                "id": "oiw223h3qndlksadffl3ksdf",
                "lastName": "Huxley",
                "updated": 2017-07-01T01:16:43.881Z,
              },
              "created": 2018-01-16T01:36:26.220Z,
              "id": "eRpcQ5H3rsTXfmh9bMvs",
              "title": "The Doors of Perception",
              "updated": 2018-01-16T01:36:26.220Z,
            },
            "onClick": [Function],
            "showAuthor": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
